### ISSUE TITLE
Opening an archived card gives a time-out

### ISSUE THREAD

**Issue by geppyz** [opfab/operatorfabric-core#6006]
In the use-case that Operator Fabric has a very big amount of cards stored in the archive, opening a archived card causes a time-out. This can happen for example when there is multiple gigabytes of archived cards. 

One of the functions that is called in the backend causing this issue is:
```JAVA
public Flux<ArchivedCardConsultationData> findByParentCard(ReactiveMongoTemplate template, ArchivedCardConsultationData parentCard) {

        Query query = new Query();
        if (parentCard.getDeletionDate() == null) {
            query.addCriteria(
                new Criteria().andOperator(
                    where(PARENT_CARD_ID_FIELD).is(parentCard.getProcess() + "." + parentCard.getProcessInstanceId()),
                    where(DELETION_DATE_FIELD).isNull()
                )
            );
        } else if (parentCard.getDeletionDate().toEpochMilli() == 0) {
            // use to exclude old card inserted in archives before adding the current feature 
            return Flux.empty();
        } else {
            query.addCriteria(
                new Criteria().andOperator(
                    where(PARENT_CARD_ID_FIELD).is(parentCard.getProcess() + "." + parentCard.getProcessInstanceId()),
                    where(PUBLISH_DATE_FIELD).lt(parentCard.getDeletionDate()),
                    new Criteria().orOperator(
                        where(DELETION_DATE_FIELD).isNull(), // when parent has keepChildCard=true
                        where(DELETION_DATE_FIELD).gte(parentCard.getDeletionDate())
                    )
                )
            );
        }
        return template.find(query,  ArchivedCardConsultationData.class);
    }
```

It requests `PARENT_CARD_ID_FIELD`, which can be translated to the field `parentCardId` in the db.archivedCards collection. This field does not seem to have an index present and is the root-cause of the issue.

I checked this by running `db.archivedCards.getIndexes()` on the database.

```JSON
[
  {
    v: 4,
    key: { _id: 1 },
    name: '_id_',
    ns: 'db.archivedCards'
  },
  {
    v: 4,
    key: { process: 1 },
    name: 'process',
    ns: 'db.archivedCards'
  },
  {
    v: 4,
    key: { state: 1 },
    name: 'state',
    ns: 'db.archivedCards'
  },
  {
    v: 4,
    key: { publishDate: 1 },
    name: 'publishDate',
    ns: 'db.archivedCards'
  },
  {
    v: 4,
    key: { startDate: 1 },
    name: 'startDate',
    ns: 'db.archivedCards'
  },
  {
    v: 4,
    key: { endDate: 1 },
    name: 'endDate',
    ns: 'db.archivedCards'
  },
  {
    v: 4,
    key: { tags: 1 },
    name: 'tags',
    ns: 'db.archivedCards'
  },
  {
    v: 4,
    key: { userRecipients: 1 },
    name: 'userRecipients',
    ns: 'db.archivedCards'
  },
  {
    v: 4,
    key: { groupRecipients: 1 },
    name: 'groupRecipients',
    ns: 'db.archivedCards'
  },
  {
    v: 4,
    key: { entityRecipients: 1 },
    name: 'entityRecipients',
    ns: 'db.archivedCards'
  },
  {
    v: 4,
    key: { processStateKey: 1 },
    name: 'processStateKey',
    ns: 'db.archivedCards'
  },
  {
    v: 4,
    key: { deletionDate: 1 },
    name: 'deletionDate',
    ns: 'db.archivedCards'
  },
  {
    v: 4,
    key: { titleTranslated: 1 },
    name: 'titleTranslated',
    ns: 'db.archivedCards'
  },
  {
    v: 4,
    key: { summaryTranslated: 1 },
    name: 'summaryTranslated',
    ns: 'db.archivedCards'
  },
  {
    v: 4,
    key: { expirationDate: 1 },
    name: 'expirationDate',
    ns: 'db.archivedCards'
  },
  {
    v: 4,
    key: { processInstanceId: 1 },
    name: 'processInstanceId_1',
    ns: 'db.archivedCards'
  }
]
```

**Comment by geppyz** [comment]
In order to fix this issue, you can run the following command on the database to create the missing index:
```BASH
db.archivedCards.createIndex({ parentCardId: 1 });
```

Reminder, if your database is really large, this will take multiple hours to complete! 
However results are great: loading a single archived card went from 1+ hours, to 2ms.

**Comment by geppyz** [comment]
Maybe this index can be added to be created automatically by Operator Fabric?

**Comment by geppyz** [comment]
This issue is fixed in coming 4.2.0 release.
